pkg/serverless/serverless.go

```diff
@@ -161,6 +161,9 @@ func callInvocationHandler(daemon *daemon.Daemon, arn string, deadlineMs int64,
 	cpuOffsetData, cpuOffsetErr := proc.GetCPUData()
 	uptimeOffset, uptimeOffsetErr := proc.GetUptime()
 	networkOffsetData, networkOffsetErr := proc.GetNetworkData()
+	sendTmpMetrics := make(chan bool)
+	go metrics.SendTmpEnhancedMetrics(sendTmpMetrics, daemon.ExtraTags.Tags, daemon.MetricAgent)
```

Comment:
Idea for later.

There's an awful lot of setup code here for all these new enhanced metrics. Might clean things up to create a new package with its own type that manages all of this. Something with an interface like:

```go
func callInvocationHandler(daemon *daemon.Daemon, arn string, deadlineMs int64, safetyBufferTimeout time.Duration, requestID string, invocationHandler InvocationHandler) {
	sysMetrics := sysmetrics.NewSystemMetrics()

	timeout := computeTimeout(time.Now(), deadlineMs, safetyBufferTimeout)
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	defer cancel()
	doneChannel := make(chan bool)
	daemon.TellDaemonRuntimeStarted()
	go invocationHandler(doneChannel, daemon, arn, requestID)
	select {
	case <-ctx.Done():
		log.Debug("Timeout detected, finishing the current invocation now to allow receiving the SHUTDOWN event")
		// Tell the Daemon that the runtime is done (even though it isn't, because it's timing out) so that we can receive the SHUTDOWN event
		daemon.TellDaemonRuntimeDone()
		break
	case <-doneChannel:
		break
	}

	sysMetrics.Send()
}
```

This way, this `callInvocationHandler` function doesn't need to have any knowledge of how the system metrics are created. All it will need to do is set them up and then tell them to send.

Just an idea.

Reply:
haha, and if you wanted to be real clever you could do something like

```go
func callInvocationHandler( ... ) {
	defer sysmetrics.NewSystemMetrics().Send()

	...
}
```

But that might be too confusing for the reader.

Reply:
I like it! I'll give a try in another pass